### PR TITLE
fix  dos new line escape

### PR DESCRIPTION
**Pull request by JoseConseco**
fixes: https://github.com/dyng/ctrlsf.vim/issues/299
hopefully without affecting vanilla vim.

### CODE REVIEW

File: autoload/ctrlsf/class/line.vim [hunk 1]
```diff
@@ -18,7 +18,7 @@ func! ctrlsf#class#line#New(fname, lnum, content) abort
     return {
         \ 'lnum'      : a:lnum,
         \ 'vpos'      : {'normal':{'lnum':-1}},
-        \ 'content'   : a:content,
+        \ 'content'   : substitute(a:content, "\r$", "", ""),
```
Comment (dyng) [inline autoload/ctrlsf/class/line.vim:21]: It's better to fix in `ctrlsf#db#ParseBackendResult`. The result is split into lines by line feed and have no trailing `\r`.

Comment (dyng) [reply autoload/ctrlsf/class/line.vim:21]: You can print debug logs by `let g:ctrlsf_debug_mode=1` and see why substitution is not performed.

Comment (dyng) [reply autoload/ctrlsf/class/line.vim:21]: You will see logs like

```
DefactorizeLine: [Factor]: [/Users/dingye/Dropbox/Projects/CtrlSf/Rakefile, 1, task :archive do], [Orig]: /Users/dingye/Dropbox/Projects/CtrlSf/Rakefile-1-task
```

Please collect them and comment here. I need to confirm my hypothesis that `split` is the root cause for your issue.

Comment (JoseConseco) [reply autoload/ctrlsf/class/line.vim:21]: Ok here is result of search for 'GTool_OT_RayCastEditPins' : 
image
And after search is performed:
image
I'm using  `<Plug>CtrlSFCCwordPath<CR>` command for search. Too bad there seem to be no  vimscript dap server for nvim integration.

**Comment by dyng** [comment]
Thanks to your debug logs, I have nailed it. Please try branch `fix/cr-problem` and test if your issue is fixed. @JoseConseco

**Comment by JoseConseco** [comment]
I just tested new branch and that indeed fixed the issue! Thanks @dyng

**Comment by dyng** [comment]
`fix/cr-problem` has been merged into master. Thanks for your contribution! I'll close this pull request as the issue is fixed. @JoseConseco

**Pull request closed without merging**